### PR TITLE
ScriptingPython: portable version binds to Python library 2.7-3.13 at runtime

### PR DESCRIPTION
**Pull request by tuffnatty**
This PR is assumed to be applied on top of #5047 and #5048.

It adds support for runtime binding to Python library configured by user. It also adds some needed UI features.

Screenshot 2024-08-19 at 04 22 41

Tested on: MacOS 10.14, Python SDK 3.4–3.13, Python library 2.7–3.13.
Fixes: #4662.

### CODE REVIEW

**Comment by tuffnatty** [comment]
I am not sure why CodeFactor reports additional code complexity in scriptingpython.cpp (no line number). Yes, there is a preprocessor-generated function body with lots of `if`s, but it's in another module, dynamicapythonapi.cpp.

File: Plugins/ScriptingPython/scriptingpython.cpp [hunk 1]
```diff
@@ -40,33 +67,196 @@ ScriptingPython::~ScriptingPython()
...
+static void globParts(QStringList& files, QList<QStringList> parts,
+                      QString prefix = QString(), int offset = 0)
+{
+    // Find all existing files among all concatenations of path parts.
+    // A part may include wildcards or path separators but not both.
+    if (parts.size() <= offset)
+        return;
+    bool isFile = offset == parts.size() - 1;
+    QDir::Filters filters = isFile ? QDir::Files : QDir::Dirs | QDir::NoDotAndDotDot | QDir::NoSymLinks;
+    for (QString piece : parts.at(offset))
```
Comment (pawelsalawa) [inline Plugins/ScriptingPython/scriptingpython.cpp:183]: I think the "complexity" issue comes from here. It's a long, multiline body of the `for` loop, while it's not enclosed in braces. Especially that the body itself contains another multiline `if`s. It has potential for breaking the code more easily in future. Usually I allow myself to skip braces only if it's a single line body and if it's the last level of `if` or `loop`. So far did not get complaints about that.

(btw. it's the one of 2 reasons, for which I don't like Python - making indentation essential for code semantics.)

**Comment by pawelsalawa** [comment]
General comment - it's an awesome improvement!

File: Plugins/ScriptingPython/scriptingpython.cpp (hunk 1 above)
Comment (tuffnatty) [reply Plugins/ScriptingPython/scriptingpython.cpp:183]: Sure, I'll add the cosmetic braces. I've forgot about this piece.

**Comment by tuffnatty** [comment]
Added some cosmetic braces and comments.

**Pull request merged**